src/LLMProviders/chatModelManager.ts

```diff
@@ -108,15 +108,31 @@ export default class ChatModelManager {
       },
     };
 
-    return { ...baseConfig, ...(providerConfig[chatModelProvider as keyof typeof providerConfig] || {}) };
+    const selectedProviderConfig =
+      providerConfig[chatModelProvider as keyof typeof providerConfig] || {};
+
+    // When useOpenAILocalProxy is enabled, use local proxy server
+    // Local proxy server will proxy requests to openAIProxyBaseUrl
+    if (
+      chatModelProvider === ModelProviders.OPENAI &&
+      params.useOpenAILocalProxy &&
+      params.openAIProxyBaseUrl
+    ) {
+      (
+        selectedProviderConfig as (typeof providerConfig)[ModelProviders.OPENAI]
+      ).openAIProxyBaseUrl = `http://localhost:${PROXY_SERVER_PORT}`;
+    }
+
+    return { ...baseConfig, ...selectedProviderConfig };
```

Comment:
This could possibly be made more flexible to support a local proxy server for any model.

It looks like more providers support `openAIProxyBaseUrl`. We could enable the local proxy for any provider that supports `openAIProxyBaseUrl` or a switch/if statement to manuallu set the url to the proxy server when it's another key, like `anthropicApiUrl` and have the proxy server proxy to the respective URL

Reply:
I didn't take this route because my needs were narrow and I don't have any other subs / keys to models to verify that they work